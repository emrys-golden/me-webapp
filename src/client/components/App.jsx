import React, { Suspense, lazy } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import routes from '../routes';

const App = () => (
  <Router>
    <Suspense fallback={<div>Loading...</div>}>
      <Switch>
        {
          routes.map(route => <Route key={route.path} {...route} />)
        }
      </Switch>
    </Suspense>
  </Router>
);

export default App;
