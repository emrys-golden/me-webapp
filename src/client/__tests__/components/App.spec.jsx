import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

jest.mock('react', () => {
  const React = jest.requireActual('react');
  React.Suspense = ({ children }) => children;
  React.Suspense.displayName = 'Suspense';
  return React;
});

jest.mock('../../routes', () => ([
  {
    path: '/test',
    component: () => 'Test',
  },
]))

describe('App component tests', () => {
  it('should render correctly', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
