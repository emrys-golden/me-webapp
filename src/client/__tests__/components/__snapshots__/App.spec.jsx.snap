// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests should render correctly 1`] = `
<BrowserRouter>
  <Suspense
    fallback={
      <div>
        Loading...
      </div>
    }
  >
    <Switch>
      <Route
        component={[Function]}
        key="/test"
        path="/test"
      />
    </Switch>
  </Suspense>
</BrowserRouter>
`;
